include/zephyr/cache.h

```diff
@@ -306,7 +306,7 @@ static ALWAYS_INLINE int z_impl_sys_cache_data_invd_range(void *addr, size_t siz
  */
 static ALWAYS_INLINE int sys_cache_instr_invd_range(void *addr, size_t size)
 {
-#if defined(CONFIG_CACHE_MANAGEMENT) && defined(CONFIG_ICACHE)
+#if defined(CONFIG_CACHE_MANAGEMENT) && defined(CONFIG_ICACHE) && defined(CONFIG_ICACHE_RANGE_OPS)
 	return cache_instr_invd_range(addr, size);
 #endif
```

Comment:
I don't think it is needed to introduce a new Kconfig item "CONFIG_ICACHE_RANGE_OPS", there are already some drivers which also don't support such operation, but they defined a simple function just return -ENOTSUP (for example: [cache_andes.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/cache/cache_andes.c#L486)
And the header file of cache: include/zephyr/drivers/cache.h provides all the definitions of all these APIs, maybe there will be no building issue although the driver don't implement all of them, but I think we need to align the header file and driver.